testproject/Assets/Tests/Runtime/MultiprocessRuntime/TestCoordinator.cs

```diff
@@ -0,0 +1,337 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using MLAPI;
+using MLAPI.Configuration;
+using MLAPI.Logging;
+using MLAPI.Messaging;
+using NUnit.Framework;
+using NUnit.Framework.Interfaces;
+using UnityEngine;
+using UnityEngine.TestTools;
+using Debug = UnityEngine.Debug;
+
+/// <summary>
+/// TestCoordinator
+/// Used for coordinating multiprocess end to end tests. Used to call RPCs on other nodes and gather results
+/// This is needed to coordinate server and client execution steps. The current remote player test runner hardcodes test
+/// to run in a bootstrap scene before launching the player and doesn't call each tests individually. There's not opportunity
+/// to coordinate test execution between client and server with that model.
+/// The only per tests communication already existing is to get the results per test as they are running
+/// With this test coordinator, it's not possible to start a main test node with the test runner and have that server start other worker nodes
+/// on which to execute client tests. We use MLAPI as both a test framework and as the target of our performance tests.
+/// </summary>
+[RequireComponent(typeof(NetworkObject))]
+public class TestCoordinator : NetworkBehaviour
+{
+    public const int perTestTimeout = 5 * 60; // seconds
+
+    public const float maxWaitTimeout = 20;
+    private const char k_MethodFullNameSplitChar = '@';
+
+
+    private bool m_ShouldShutdown;
+    private float m_TimeSinceLastConnected;
+    private float m_TimeSinceLastKeepAlive;
+
+    public static TestCoordinator Instance;
+
+    private Dictionary<ulong, List<float>> m_TestResultsLocal = new Dictionary<ulong, List<float>>(); // this isn't super efficient, but since it's used for signaling around the tests, shouldn't be too bad
+    private Dictionary<ulong, bool> m_ClientIsFinished = new Dictionary<ulong, bool>();
+
+    public static Dictionary<ulong, List<float>>.KeyCollection AllClientIdsWithResults => Instance.m_TestResultsLocal.Keys;
```

Comment:
I don't think a KeyCollection is a commonly used type, maybe something a bit simpler? There should be an easy equivalent to use.

Reply:
lol indeed, switched to list<ulong>